habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -825,3 +825,128 @@ def get_object_regions(
     )
 
     return sim.semantic_scene.get_regions_for_points(key_points)
+
+
+def get_link_normalized_joint_position(
+    objectA: habitat_sim.physics.ManagedArticulatedObject, link_ix: int
+) -> float:
+    """
+    Normalize the joint limit range [min, max] -> [0,1] and return the current joint state in this range.
+
+    :param objectA: The parent ArticulatedObject of the link.
+    :param link_ix: The index of the link within the parent object. Not the link's object_id.
+
+    :return: normalized joint position [0,1]
+    """
+
+    assert objectA.get_link_joint_type(link_ix) in [
+        habitat_sim.physics.JointType.Revolute,
+        habitat_sim.physics.JointType.Prismatic,
+    ], f"Invalid joint type '{objectA.get_link_joint_type(link_ix)}'. Open/closed not a valid check for multi-dimensional or fixed joints."
+
+    joint_pos_ix = objectA.get_link_joint_pos_offset(link_ix)
+    joint_pos = objectA.joint_positions[joint_pos_ix]
+    limits = objectA.joint_position_limits
+
+    # compute the normalized position [0,1]
+    n_pos = (joint_pos - limits[0][joint_pos_ix]) / (
+        limits[1][joint_pos_ix] - limits[0][joint_pos_ix]
+    )
+    return n_pos
+
+
+def set_link_normalized_joint_position(
+    objectA: habitat_sim.physics.ManagedArticulatedObject,
+    link_ix: int,
+    normalized_pos: float,
```

Comment:
What happens if I pass some value out of range here? Will the joint state wrap around?

Reply:
Great call-out. I should assert the range [0,1] or clamp here. 
It won't wrap around, it will extrapolate past the joint limits until a clamp is performed (e.g. if auto-clamp joint limits is set a sim step will snap it back).